client/allocrunnerv2/alloc_runner.go

```diff
@@ -55,12 +54,12 @@ type allocRunner struct {
 // NewAllocRunner returns a new allocation runner.
 func NewAllocRunner(config *Config) *allocRunner {
 	ar := &allocRunner{
-		alloc:        config.Alloc,
-		clientConfig: config.ClientConfig,
-		tasks:        make(map[string]*taskrunner.TaskRunner),
-		waitCh:       make(chan struct{}),
-		updateCh:     make(chan *structs.Allocation),
-		stateDB:      config.StateDB,
+		config:   config,
```

Comment:
I'm not a fan of storing the whole struct *and* plucking some fields off of it. It makes it ambiguous which version of the field to use.

This is why I started unpacking the Config struct in New funcs. It's a little bit extra code, but it removes any potential for confusion.

It's not a hill I'm going to die on, but I do want to pick an idiom and keep it consistent.